torch/lib/THPP/Tensor.hpp

```diff
@@ -36,6 +36,8 @@ struct Tensor {
   virtual bool isContiguous() const = 0;
   virtual void* data() = 0;
   virtual const void* data() const = 0;
+  virtual void* cdata() = 0;
+  virtual const void* cdata() const = 0;
```

Comment:
Yeah I was hoping to do that eventually, but someone didn't want to review a 3,000 line diff :P 

I'd like to get things Conv and BatchNorm working with the autograd refactor before changing the cuDNN bindings.